node/src/accountant/scanners/mod.rs

```diff
@@ -850,21 +860,12 @@ impl Scanner<RetrieveTransactions, ReceivedPayments> for ReceivableScanner {
         logger: &Logger,
     ) -> Option<NodeToUiMessage> {
         if message.payments.is_empty() {
-            info!(
+            self.update_start_block_while_no_received_transactions_found(
+                message.new_start_block,
                 logger,
-                "No new received payments were detected during the scanning process."
             )
         } else {
-            let total_newly_paid_receivable = message
-                .payments
-                .iter()
-                .fold(0, |so_far, now| so_far + now.wei_amount);
-            self.receivable_dao
-                .as_mut()
-                .more_money_received(message.timestamp, message.payments);
-            self.financial_statistics
-                .borrow_mut()
-                .total_paid_receivable_wei += total_newly_paid_receivable;
+            self.update_account_balances_start_block_and_stats(&message, logger)
```

Comment:
Maybe you can write a function named `update_start_block()` and another named `update_balances_and_stats()`. That way you can call the `update_start_block()` inside the if clause and the combination of both functions in the else clause.

Reply:
it uses a completely different function call inside. Definitely not repetition. I'll think a while about the names but otherwise I cannot serve you.